lib/buffer.js

```diff
@@ -1021,557 +1002,786 @@ Buffer.prototype.slice = function slice(start, end) {
-  const x0 = this.readUInt32BE(offset + 4, noAssert);
-  return toDouble(x0, x1);
-};
-
-
-Buffer.prototype.readDoubleLE = function(offset, noAssert) {
-  offset = offset >>> 0;
-  const x0 = this.readUInt32LE(offset + 0, noAssert);
-  const x1 = this.readUInt32LE(offset + 4, noAssert);
-  return toDouble(x0, x1);
-};
-
-
-Buffer.prototype.readFloatBE = function(offset, noAssert) {
-  offset = offset >>> 0;
-  return toFloat(this.readUInt32BE(offset, noAssert));
+  var val = this[offset + 1] | (this[offset] << 8);
+  return (val & 0x8000) ? val | 0xFFFF0000 : val;
 };
 
+// Read floats
+function readFloatBackwards(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 3];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  uInt8Float32Array[3] = this[offset++];
+  uInt8Float32Array[2] = this[offset++];
+  uInt8Float32Array[1] = this[offset++];
+  uInt8Float32Array[0] = this[offset];
+  return float32Array[0];
+}
 
-Buffer.prototype.readFloatLE = function(offset, noAssert) {
-  offset = offset >>> 0;
-  return toFloat(this.readUInt32LE(offset, noAssert));
-};
-
+function readFloatForwards(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 3];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  uInt8Float32Array[0] = this[offset++];
+  uInt8Float32Array[1] = this[offset++];
+  uInt8Float32Array[2] = this[offset++];
+  uInt8Float32Array[3] = this[offset];
+  return float32Array[0];
+}
 
-function checkInt(buffer, value, offset, ext, max, min) {
-  if (value > max || value < min)
-    throw new errors.RangeError('ERR_INVALID_OPT_VALUE', 'value', value);
-  if (offset + ext > buffer.length)
-    throw new errors.RangeError('ERR_INDEX_OUT_OF_RANGE');
+function readDoubleBackwards(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 7];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  uInt8Float64Array[7] = this[offset++];
+  uInt8Float64Array[6] = this[offset++];
+  uInt8Float64Array[5] = this[offset++];
+  uInt8Float64Array[4] = this[offset++];
+  uInt8Float64Array[3] = this[offset++];
+  uInt8Float64Array[2] = this[offset++];
+  uInt8Float64Array[1] = this[offset++];
+  uInt8Float64Array[0] = this[offset];
+  return float64Array[0];
 }
 
+function readDoubleForwards(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 7];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  uInt8Float64Array[0] = this[offset++];
+  uInt8Float64Array[1] = this[offset++];
+  uInt8Float64Array[2] = this[offset++];
+  uInt8Float64Array[3] = this[offset++];
+  uInt8Float64Array[4] = this[offset++];
+  uInt8Float64Array[5] = this[offset++];
+  uInt8Float64Array[6] = this[offset++];
+  uInt8Float64Array[7] = this[offset];
+  return float64Array[0];
+}
 
+// Write integers
```

Comment:
Pointless comment.

Reply:
When I scroll through this code I always had to check what part it was about and where the next part started. That is why I added those comments. I do not feel strongly about them but I think they make it a tiny bit clearer while scrolling where you are.